packages/graph/hash_graph/lib/graph/src/types/schema/mod.rs

```diff
@@ -17,15 +17,15 @@ use core::fmt;
 
 #[doc(inline)]
 pub use self::{data_type::DataType, entity_type::EntityType, property_type::PropertyType};
-use crate::types::Uri;
+use crate::types::{BaseUri, VersionedUri};
 
 #[derive(Debug)]
 pub enum ValidationError {
-    /// A schema has marked a property with a [`Uri`] as required but the [`Uri`] does not exist in
-    /// the `properties`.
-    MissingRequiredProperty(Uri),
+    /// A schema has marked a property with a [`BaseUri`] as required but the [`BaseUri`] does not
+    /// exist in the `properties`.
+    MissingRequiredProperty(BaseUri),
```

Comment:
Maybe we should consider using `thiserror` here so we don't have to duplicate the explanation in the doc-comment and also then in the `Display` implementation. So we'd instead just have something like:

```rust
    #[error("the schema has marked the `{0}` property as required but it wasn't defined in the `properties` object")]
    MissingRequiredProperty(String),
```

Reply:
We could use `thiserror`, but this would forbid us from implementing `Context::provide` if we want to later on.

Reply:
I think for simple error types we can perhaps use `thiserror` and then use something else if we ever need to provide. But yeah lets keep that in mind and hold off until we revisit